manifests/service.pp

```diff
@@ -109,6 +109,14 @@
       content => template('logstash/logstash.yml.erb'),
     }
 
+    if defined(Class['java']) {
+      Class['java'] -> Exec['logstash-system-install']
+
+      $use_environment = ["JAVA_HOME=${java::use_java_home}"]
```

Comment:
This is breaking encapsulation by reaching inside the implementation of another class by another author.